libs/vm/include/vm/module.hpp

```diff
@@ -179,31 +198,36 @@ class Module
       return *this;
     }
 
-    template <typename GetterReturnType, typename... GetterArgs, typename SetterReturnType,
-              typename... SetterArgs>
-    ClassInterface &InternalEnableIndexOperator(GetterReturnType (Type::*getter)(GetterArgs...),
-                                                SetterReturnType (Type::*setter)(SetterArgs...))
+    template <typename Getter, typename Setter>
+    ClassInterface &InternalEnableIndexOperator(Getter const &getter, Setter const &setter)
```

Comment:
Here and below, is there any special sense in exactly `const &` templated arguments? It automatically decays template parameters but automatically rejects rvalues.

Reply:
Right now the way we're using the callables is by capturing them in lambdas, which happens by value anyway. I don't think it makes a difference until we overcome this limitation.

Reply:
So, for instance, by-value can be facilitated if we have it as rvalue; lvalues mandate copying.
I'm not saying it is exactly required but if we have two options that work approx. the same on current codebase but one of them is in theory better than the other, why go with the other?

Reply:
If you put it like that, I suppose it makes sense to pass by value.

Reply:
Quite reasonable.